message/infrastructure/nats/publisher.go

```diff
@@ -8,15 +8,7 @@ import (
 )
 
 type StreamingPublisherConfig struct {
-	// ClusterID is the NATS Streaming cluster ID.
-	ClusterID string
-
-	// ClientID is the NATS Streaming client ID to connect with.
-	// ClientID can contain only alphanumeric and `-` or `_` characters.
-	ClientID string
-
-	// StanOptions are custom options for a connection.
-	StanOptions []stan.Option
+	NatsStreamingConfig
```

Comment:
IMO it would be better to not make breaking change here, but just pass `Marshaler` to  `NewStreamingPublisherWithStanConn`  and don't touch `NewStreamingPublisher`

Reply:
fixed